### PR TITLE
Mentor form Redesign

### PR DESCRIPTION
**Pull request by Isha988**
## Related Issue
Closes: #268 

## Description of Changes
Changed the mentor registration form into multi step form.

### CODE REVIEW

**Comment by Isha988** [comment]
@anmode 
1. Try to merge PR #255 as soon as possible so that I can add those changes into form.
2. I want to do form validation on frontend for that please guide me on "how to do form validation of multi step form" 

Till then, have a look at improved version of registration form.

**Comment by muhammadjufry** [comment]
@Isha988 Sure we will merge the pr soon(today)

**Review by anmode — APPROVED** [review]
Impressive @Isha988. I really like the component structure that you created.

**Comment by anmode** [comment]
> 2\. I want to do form validation on frontend for that please guide me on "how to do form validation of multi step form"

Sure, 
for your 2nd question As of now I know using Jquery we can do that. but that is not the best approach I think. There should be a better library like `React Hook Form` or  `veeValidate` I previously worked with veeValidate it is awesome. I'll revert back with best things we can do as of now

**Comment by anmode** [comment]
@Isha988 I have better option:https://github.com/anmode/grabtern-frontend/pull/255
could you include the changes that you want from this pR as you will then have to fight with many conflicts :(

**Comment by Isha988** [comment]
> @Isha988 I have better option:#255 could you include the changes that you want(mentor form date picker) from this pR otherwise. you have to fight with many conflicts :(
> 
> so we will merge your PR first ( mentor form is complete and then we will merge our pR that is for session datepicker)

Okay, I will add take the code from that pr and will add into mine but I am thinking of doing form validation in another pr after both of these get merged.

And please tell me once again, what we exactly want in last step of form that's "tell us about you schedule", I am bit confused.

**Comment by anmode** [comment]
@Isha988 if you face any issues please let me know. Ig as of now you just have to copy-paste the code and this PR will be completed soon and good to merge.

**Comment by Isha988** [comment]
Schedule step of mentor form has been updated.

**Comment by anmode** [comment]
I see merging this for now! Will Do further improvement in another PR.

**Review by anmode — APPROVED** [review]
Cool! There are issues.. will see them in another PR. Thank you @Isha988

File: components/MentorRegistration/SessionDetails.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+          <option value="monday">Monday</option>
+          <option value="tuesday">Tuesday</option>
+          <option value="wednesday">Wednesday</option>
+          <option value="thursday">Thursday</option>
+          <option value="friday">Friday</option>
+          <option value="saturday">Saturday</option>
+          <option value="sunday">Sunday</option>
+        </select>
+      </div>
+      {/* day input ends */}
```
Comment (anmode) [inline components/MentorRegistration/SessionDetails.js:89]: for another PR can we have component of this too?

File: components/MentorRegistration/SessionDetails.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+          </option>
+          <option value="(GMT-11:00) Pacific/Niue">
+            (GMT-11:00) Pacific/Niue
+          </option>
+          <option value="(GMT-11:00) Pacific/Pago_Pago">
+            (GMT-11:00) Pacific/Pago_Pago
+          </option>
+          <option value="(GMT-10:00) Pacific/Honolulu">
+            (GMT-10:00) Pacific/Honolulu
+          </option>
```
Comment (anmode) [inline components/MentorRegistration/SessionDetails.js:111]: same for this? so we can use this else where

**Review by anmode — APPROVED** [review]
All property of mentor form should be there in seperate css file inside `styles` folder

**Comment by anmode** [comment]
Please create a follow-up PR!  thanks for this.

**Comment by Isha988** [comment]
> Please create a follow-up PR! thanks for this.

should I first create a issue for that?

**Comment by anmode** [comment]
> > Please create a follow-up PR! thanks for this.
> 
> should I first create a issue for that?

Direct PR is ok! Just tag this pR id

**Pull request merged**